### PR TITLE
Check for server's range request support before loading remote copc point cloud

### PR DESCRIPTION
**Pull request by NEDJIMAbelgacem**
## Description
Addresses https://github.com/qgis/QGIS/issues/48304

### CODE REVIEW

File: src/core/pointcloud/qgslazinfo.h [hunk 1]
```diff
@@ -130,10 +135,8 @@ class CORE_EXPORT QgsLazInfo
     //! Static function to create a QgsLazInfo class from a file over network
     static QgsLazInfo fromUrl( QUrl &url );
 
-#ifndef SIP_RUN
-    //! Returns the LAZPERF header object
-    lazperf::header14 header() const { return mHeader; }
-#endif
+    //! Static functions to check whether the server of URL \a url supports range queries
```
Comment (uclaros) [inline src/core/pointcloud/qgslazinfo.h:138]:
```suggestion
    //! Static function to check whether the server of URL \a url supports range queries
```

File: src/core/providers/copc/qgscopcprovider.cpp [hunk 2]
```diff
@@ -25,6 +25,7 @@
 #include "qgsapplication.h"
 #include "qgsprovidersublayerdetails.h"
 #include "qgsproviderutils.h"
+#include "qgsmessagelog.h"
```
Comment (uclaros) [inline src/core/providers/copc/qgscopcprovider.cpp:28]: leftover?

File: src/core/providers/copc/qgscopcprovider.cpp [hunk 3]
```diff
@@ -33,13 +34,16 @@
 #define PROVIDER_KEY QStringLiteral( "copc" )
 #define PROVIDER_DESCRIPTION QStringLiteral( "COPC point cloud data provider" )
 
+#include <QDebug>
```
Comment (uclaros) [inline src/core/providers/copc/qgscopcprovider.cpp:37]: leftover?

File: src/core/pointcloud/qgslazinfo.cpp [hunk 4]
```diff
@@ -17,6 +17,8 @@
 
 #include "qgslazinfo.h"
 
+#include <QEventLoop>
+
```
Comment (uclaros) [inline src/core/pointcloud/qgslazinfo.cpp:21]: leftover?

File: src/core/pointcloud/qgseptpointcloudindex.cpp [hunk 5]
```diff
@@ -446,6 +447,7 @@ bool QgsEptPointCloudIndex::loadHierarchy()
     if ( errH.error != QJsonParseError::NoError )
     {
       QgsDebugMsgLevel( QStringLiteral( "QJsonParseError when reading hierarchy from file %1" ).arg( filename ), 2 );
+      mError = QStringLiteral( "QJsonParseError when reading hierarchy from file %1" ).arg( filename );
```
Comment (uclaros) [inline src/core/pointcloud/qgseptpointcloudindex.cpp:450]: While this description is ok for the message log, it seems a bit too technical for the user.

Comment (NEDJIMAbelgacem) [reply src/core/pointcloud/qgseptpointcloudindex.cpp:450]: I prefer to have a too technical error message than saying that the point cloud just didn't load and not indicate what went wrong to the user + this message has the file name so it is easier to figure out what went wrong

**Review by uclaros — APPROVED** [review]

**Pull request merged**